test/fail/timer-no-send.mo

```diff
@@ -0,0 +1,26 @@
+import { setTimer } = "mo:⛔";
+
+// no send capability: must not call `setTimer`
+func bowm() {
+    ignore setTimer(1_000_000, false, func () : async () { });
+};
+
+// transferred send capability: may call `setTimer`
+func gawd<async>() {
+    ignore setTimer(1_000_000, false, func () : async () { });
+
+    ignore async 42 // not allowed
+};
+
+// these are allowed to contain sends
+func eeek<async>() : async () {
```

Comment:
Should we disallow this combination?

Reply:
It appears harmless:
``` Motoko
Motoko compiler (source 0.10.4-27-g9a65ab1ca-dirty)
> func foo<async, T>(u : T) : async () {};
let foo : <T>T -> async () = func
```

Reply:
I reckon the parser is just add idempotently adding the initial scope parameter. That's probably how it worked when I had similar explicit syntax for this.

Reply:
Right, it checks for its presence before adding a scope entry.